src/Service/SignalsService.php

```diff
@@ -0,0 +1,264 @@
+     * @param DatasourceEntity $datasourceEntity
+     * @param MetadataEntity $entity
+     * @param QueryTime $from
+     * @param QueryTime $until
+     * @param int $step
+     * @return string influx query string
+     * @throws BackendException
+     */
+    private function buildInfluxQuery(DatasourceEntity $datasourceEntity, MetadataEntity $entity,
+                                      QueryTime $from, QueryTime $until, int $step): string {
+        $entityType = $entity->getType()->getType();
+        $entityCode = $entity->getCode();
+        if($datasourceEntity->getDatasource() == "ucsd-nt"){
+            // create query template
+            if($entityType == "country"){
+                $template = "q=SELECT mean(\"uniq_src_ip\") FROM \"geo_country\" WHERE (\"telescope\" = 'ucsd-nt' AND \"country_code\" = '%s' AND \"filter\" = 'non-erratic' AND \"geo_db\" = 'netacuity') AND %s GROUP BY time(%ds) fill(null)";
+            } else if($entityType == "region"){
+                $template = "q=SELECT mean(\"uniq_src_ip\") FROM \"geo_region\" WHERE (\"telescope\" = 'ucsd-nt' AND \"country_code\" = '%s' AND \"filter\" = 'non-erratic' AND \"geo_db\" = 'netacuity') AND %s GROUP BY time(%ds) fill(null)";
+            } else if($entityType == "county"){
+                $template = "q=SELECT mean(\"uniq_src_ip\") FROM \"geo_county\" WHERE (\"telescope\" = 'ucsd-nt' AND \"country_code\" = '%s' AND \"filter\" = 'non-erratic' AND \"geo_db\" = 'netacuity') AND %s GROUP BY time(%ds) fill(null)";
+            } else if($entityType == "asn"){
+                $template = "q=SELECT mean(\"uniq_src_ip\")  FROM \"origin_asn\" WHERE (\"telescope\" = 'ucsd-nt' AND  \"filter\" = 'non-erratic' AND \"asn\" = '%s') AND %s GROUP BY time(%ds) fill(null)";
+            } else {
+                throw new BackendException("Unsupported metadata entity type: $entityType");
+            }
+            // convert epoch time to nanoseconds
+            $timeQuery = sprintf("time >= %ds AND time < %ds", $from->getEpochTime(), $until->getEpochTime());
+            return sprintf($template, $entityCode, $timeQuery, $step);
+        } else {
+            throw new BackendException(
+                sprintf("Datasource %s doesn't not currently support InfluxDB backend",
+                    $datasourceEntity->getDatasource()
+                ));
+        }
+    }
+
+
+    /**
+     * @param QueryTime $from
+     * @param QueryTime $until
+     * @param MetadataEntity $entity
+     * @param DatasourceEntity $datasource
+     * @param int $maxPoints
+     * @return TimeSeries
+     * @throws BackendException
+     */
+    public function queryForTimeSeries(QueryTime $from, QueryTime $until, MetadataEntity $entity,
+                                  DatasourceEntity $datasource, ?int $maxPoints): TimeSeries {
+        $backend = $datasource->getBackend();
+
+        if($backend == "graphite"){
+            $exp_json = $this->buildGraphiteExpression($entity, $datasource);
```

Comment:
since you're just building a query string and have drastically simplified the graphite backend, you could even throw out the Charthouse Expression stuff and instead just directly store/pass the graphite query.
this isn't a big deal, but it will let you delete a bunch more code.